core/src/main/java/org/infinispan/interceptors/impl/AsyncInterceptorChainImpl.java

```diff
@@ -223,6 +223,7 @@ public void appendInterceptor(AsyncInterceptor ci, boolean isCustom) {
    @Override
    public CompletableFuture<Object> invokeAsync(InvocationContext ctx, VisitableCommand command) {
       try {
+         command.init(componentRegistry);
```

Comment:
If we remove this, it's also necessary to call `.init` on the `PrepareCommands` modifications when `invokeAsync(ComponentRegistry)`

Reply:
Right, I saw `PrepareCommand` calling `init()` on the modifications and I didn't realize it was in the `init()` method. But it should be safe to move that code to `invokeAsync()`, because we don't expect `init()` to be called on commands that are both `CacheRpcCommand` and `VisitableCommand`.